lib/services/plugin-variables-service.ts

```diff
@@ -0,0 +1,103 @@
+"use strict";
+
+import * as helpers from "./../common/helpers";
+
+export class PluginVariablesService implements IPluginVariablesService {
+	private static PLUGIN_VARIABLES_KEY = "variables";
+
+	constructor(private $errors: IErrors,
+		private $fs: IFileSystem,
+		private $pluginVariablesHelper: IPluginVariablesHelper,
+		private $projectData: IProjectData,
+		private $projectDataService: IProjectDataService,
+		private $prompter: IPrompter,
+		private $staticConfig: IStaticConfig) { }
+
+	public savePluginVariablesInProjectFile(pluginData: IPluginData): IFuture<void> {
+		return (() => {
+			let values = Object.create(null);
+			this.executeForAllPluginVariables(pluginData, (pluginVariableData: IPluginVariableData) =>
+				(() => {
+					let pluginVariableValue = this.getPluginVariableValue(pluginVariableData).wait();
+					if(!pluginVariableValue) {
+						this.$errors.failWithoutHelp(`Unable to find value for ${pluginVariableData.name} plugin variable from ${pluginData.name} plugin. Ensure the --var option is specified or the plugin variable has default value.`);
+					}
+					values[pluginVariableData.name] = pluginVariableValue;
+				}).future<void>()()).wait();
+
+			this.$projectDataService.initialize(this.$projectData.projectDir);
+			this.$projectDataService.setValue(this.getPluginVariablePropertyName(pluginData), values).wait();
+		}).future<void>()();
+	}
+
+	public removePluginVariablesFromProjectFile(pluginData: IPluginData): IFuture<void> {
+		this.$projectDataService.initialize(this.$projectData.projectDir);
+		return this.$projectDataService.removeProperty(this.getPluginVariablePropertyName(pluginData));
+	}
+
+	public interpolatePluginVariables(pluginData: IPluginData, pluginConfigurationFileContent: string): IFuture<string> {
+		return (() => {
+			this.executeForAllPluginVariables(pluginData, (pluginVariableData: IPluginVariableData) =>
+				(() => {
+					if(!pluginVariableData.value) {
+						this.$errors.failWithoutHelp(`Unable to find the value for ${pluginVariableData.name} plugin variable into project package.json file. Verify that your package.json file is correct and try again.`);
+					}
+
+					pluginConfigurationFileContent = pluginConfigurationFileContent.replace(new RegExp(`{${pluginVariableData.name}}`, "gi"), pluginVariableData.value);
+				}).future<void>()()).wait();
+			return pluginConfigurationFileContent;
+		}).future<string>()();
+	}
+
+	private getPluginVariableValue(pluginVariableData: IPluginVariableData): IFuture<string> {
+		return (() => {
+			let pluginVariableName = pluginVariableData.name;
+			let value = this.$pluginVariablesHelper.getPluginVariableFromVarOption(pluginVariableName);
+			if(value) {
+				value = value[pluginVariableName];
+			} else {
+				value = pluginVariableData.defaultValue;
+				if(!value && helpers.isInteractive()) {
+					let promptSchema = {
+						name: pluginVariableName,
+						type: "input",
+						message: `Enter value for ${pluginVariableName} variable: `
```

Comment:
As a side note, if the user presses sets a false-like value for a plugin variable we will fail in `savePluginVariablesInProjectFile` function up above and the process will end which is no great UX.
Instead we can use `$prompter`'s  `validate` property like so 
```Typescript
validate: (value: string) => !!value ? true : 'Please enter a value!'
```